src/Microsoft.ML.Transforms/TensorflowTransform.cs

```diff
@@ -0,0 +1,665 @@
+        /// </summary>
+        private readonly int _batchSize;
+
+        /// <summary>
+        /// Convenience constructor for public facing API.
+        /// </summary>
+        /// <param name="env">Host Environment.</param>
+        /// <param name="input">Input <see cref="IDataView"/>. This is the output from previous transform or loader.</param>
+        /// <param name="modelFile">This is the frozen model file. https://www.tensorflow.org/mobile/prepare_models </param>
+        /// <param name="name">Name of the output column. Keep it same as in the Tensorflow model.</param>
+        /// <param name="source">Name of the input column(s). Keep it same as in the Tensorflow model.</param>
+        public TensorflowTransform(IHostEnvironment env, IDataView input, string modelFile, string name, params string[] source)
+            : this(env, new Arguments() { Column = new[] { new Column() { Source = source, Name = name } }, ModelFile = modelFile }, input)
+        {
+        }
+
+        public TensorflowTransform(IHostEnvironment env, Arguments args, IDataView input)
+            : base(env, RegistrationName, input)
+        {
+            Host.CheckValue(args, nameof(args));
+            Host.CheckUserArg(Utils.Size(args.Column) > 0, nameof(args.Column));
+            for (int i = 0; i < args.Column.Length; i++)
+                Host.CheckUserArg(Utils.Size(args.Column[i].Source) > 0, nameof(args.Column));
+
+            _batchSize = args.BatchSize;
+            _session = LoadTFSession(args.ModelFile);
+            _bindings = new Bindings(args.Column, Source.Schema, this);
+        }
+
+        private TensorflowTransform(IHost host, ModelLoadContext ctx, IDataView input)
+           : base(host, input)
+        {
+            Host.AssertValue(ctx);
+
+            _batchSize = ctx.Reader.ReadInt32();
+
+            byte[] data = null;
+            ctx.TryLoadBinaryStream("TFModel", r =>
+            {
+                data = r.ReadByteArray();
+            });
+
+            var graph = new TFGraph();
+            graph.Import(data);
+            _session = new TFSession(graph);
+            _bindings = new Bindings(ctx, Source.Schema, this);
+        }
+
+        ~TensorflowTransform()
+        {
+            _session.CloseSession();
+            _session.DeleteSession();
+        }
+
+        public static TensorflowTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            var h = env.Register(RegistrationName);
+            h.CheckValue(ctx, nameof(ctx));
+            h.CheckValue(input, nameof(input));
+            ctx.CheckAtModel(GetVersionInfo());
+            return h.Apply("Loading Model", ch => new TensorflowTransform(h, ctx, input));
+        }
+
+        private TFSession LoadTFSession(string modelFile)
+        {
+            var graph = new TFGraph();
+            graph.Import(File.ReadAllBytes(modelFile), "");
```

Comment:
>Import [](start = 18, length = 6)

This call can go in a try/catch, with the catch rethrowing the exception through the Host. You can add an argument to LoadTFSession to tell the Host whether to throw ExceptDecode() or ExceptUserArg().

Reply:
I think this method can be used both in the ctor from arguments and the ctor from file, and take a bool parameter indicating what kind of exception to throw.

---
In reply to: [210650887](https://github.com/zeahmed/machinelearning/pull/3#discussion_r210650887) [](ancestors = 210650887)